pkg/controller/operators/operator_controller.go

```diff
@@ -152,9 +153,11 @@ func (r *OperatorReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 			return ctrl.Result{Requeue: true}, nil
 		}
 	} else {
-		if err := r.Status().Update(ctx, operator.Operator); err != nil {
-			log.Error(err, "Could not update Operator status")
-			return ctrl.Result{Requeue: true}, nil
+		if !equality.Semantic.DeepEqual(in.Status, operator.Operator.Status) {
+			if err := r.Status().Update(ctx, operator.Operator); err != nil {
+				log.Error(err, "Could not update Operator status")
+				return ctrl.Result{Requeue: true}, nil
```

Comment:
nit: any reason we're requeuing here when we know we're in an error path?

```go
if err := r.Status().Update(ctx, operator.Operator); err != nil {
    return ctrl.Result{}, err
```

Do we need to log an error here if we're requeueing?

Reply:
In my limited experience calls to the api like this could result in ephemeral errors that may clear themselves on the next attempt. I guess it really depends on what kind of error you're getting back though.

Reply:
I think my original comment wasn't super clear: I was basically asking whether we should be specifying `ctrl.Result{Requeue: true}, nil` here vs. `ctrl.Result{}, err` given we already know we're handling a non-nil error in this conditional.

This isn't a blocker though, as both of those implementations are semantically equal given they will result in a requeue being processed. The latter is more prevalent w.r.t the OLM repositories, so that implementation would remain consistent with the rest of the OLM controlled codebases.

Reply:
And I also think an implementation that logs, and then returns an error is an anti-pattern given you can expect the call site to determine whether the error should be logged.

Reply:
Oh! Thanks for the explanation, your comment makes more sense now. I had missed the `err` vs `nil` return there on my first look through.

Reply:
That's a good point @timflannagan , I didn't change the update call from the original code but I agree that we ought to just return the error. I can make a quick update.

Reply:
I started updating this, but this pattern happens all over the operator controller, I'm gonna do this in a separate PR.

Reply:
Yeah if this is a pattern throughout the operator controller, then this implementation is fine with me for now.